transmart-oauth/src/main/groovy/org/transmart/oauth/TransmartOauthGrailsPlugin.groovy

```diff
@@ -0,0 +1,174 @@
+
+            sessionRegistry(SessionRegistryImpl)
+
+            redirectStrategy(DefaultRedirectStrategy)
+            accessDeniedHandler(AccessDeniedHandlerImpl) {
+                errorPage = '/login'
+            }
+            failureHandler(SimpleUrlAuthenticationFailureHandler) {
+                defaultFailureUrl = '/login'
+            }
+
+            transactionInterceptor(TransactionInterceptor) {
+                transactionManagerBeanName = 'transactionManager'
+                transactionAttributeSource = ref('transactionAttributeSource')
+            }
+
+            def transmartSecurity = grailsApplication.config.org.transmart.security
+            if (SpringSecurityUtils.securityConfig.ldap.active) {
+                ldapUserDetailsMapper(LdapAuthUserDetailsMapper) {
+                    springSecurityService = ref('springSecurityService')
+                    // pattern for newly created user, can include <ID> for record id or <FEDERATED_ID> for external user name
+                    if (transmartSecurity.ldap.newUsernamePattern) {
+                        newUsernamePattern = transmartSecurity.ldap.newUsernamePattern
+                    }
+                    // comma separated list of new user authorities
+                    if (transmartSecurity.ldap.defaultAuthorities) {
+                        defaultAuthorities = transmartSecurity.ldap.defaultAuthorities
+                    }
+                    // if inheritPassword == false specified user will not be able to login without LDAP
+                    inheritPassword = transmartSecurity.ldap.inheritPassword
+                    // can be 'username' or 'federatedId'
+                    mappedUsernameProperty = transmartSecurity.ldap.mappedUsernameProperty
+                }
+
+                if (grailsApplication.config.org.transmart.security.ldap.ad.domain) {
+
+                    adExtension(ActiveDirectoryLdapAuthenticationExtension)
+
+                    aop {
+                        config("proxy-target-class": true) {
+                            aspect(id: 'adExtensionService', ref: 'adExtension')
+                        }
+                    }
+
+                    ldapAuthProvider(ActiveDirectoryLdapAuthenticationProvider,
+                            transmartSecurity.ldap.ad.domain,
+                            SpringSecurityUtils.securityConfig.ldap.context.server
+                    ) {
+                        userDetailsContextMapper = ref('ldapUserDetailsMapper')
+                    }
+                }
+            }
+
+            bruteForceLoginLockService = ref('bruteForceLoginLockService')
+            bruteForceLoginLockService(BruteForceLoginLockService) {
+                allowedNumberOfAttempts = grailsApplication.config.bruteForceLoginLock.allowedNumberOfAttempts
+                lockTimeInMinutes = grailsApplication.config.bruteForceLoginLock.lockTimeInMinutes
+            }
```

Comment:
As I checked - bruteForceLoginLockService is properly injected in transmartApp, including settings for allowedNumberOfAttempts and lockTimeInMinutes (I tested it in LoginController)